azure-iot-device/tests/common/pipeline/test_pipeline_stages_base.py

```diff
@@ -715,3 +718,460 @@ def test_ignores_unknown_request_id(self, stage, op, iot_response, unhandled_err
         stage.next._send_event_up(iot_response)
         assert op.callback.call_count == 0
         assert unhandled_error_handler.call_count == 0
+
+
+"""
+A note on terms in the TimeoutStage tests:
+    No-timeout ops are ops that don't need a timeout check
+    Yes-timeout ops are ops that do need a timeout check
+"""
+timeout_intervals = {
+    pipeline_ops_mqtt.MQTTSubscribeOperation: 10,
```

Comment:
this is copy paste code (also in pipeline_stages_base). Should it be moved into it's own model or something?

Reply:
Absolutely.  I think that might be part of the retry policy object.  For know, KISS. 

---
In reply to: [337146050](https://github.com/Azure/azure-iot-sdk-python/pull/365#discussion_r337146050) [](ancestors = 337146050)